src/Datadog.Trace.ClrProfiler.Managed/Integrations/GraphQLIntegration.cs

```diff
@@ -0,0 +1,381 @@
+                {
+                    var task = (Task<T>)originalMethod(executionStrategy, executionContext);
+                    var executionResult = await task.ConfigureAwait(false);
+                    RecordExecutionErrorsIfPresent(scope.Span, "GraphQL.ExecutionError", executionContext.GetProperty("Errors").GetValueOrDefault());
+                    return executionResult;
+                }
+                catch (Exception ex)
+                {
+                    scope?.Span.SetException(ex);
+                    throw;
+                }
+            }
+        }
+
+        private static Scope CreateScopeFromValidate(object document)
+        {
+            if (!Tracer.Instance.Settings.IsIntegrationEnabled(IntegrationName))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            Tracer tracer = Tracer.Instance;
+            string source = document.GetProperty<string>("OriginalQuery")
+                                    .GetValueOrDefault();
+            string serviceName = string.Join("-", tracer.DefaultServiceName, ServiceName);
+
+            Scope scope = null;
+
+            try
+            {
+                scope = tracer.StartActive(ValidateOperationName, serviceName: serviceName);
+                var span = scope.Span;
+                span.Type = SpanTypes.GraphQL;
+
+                span.SetTag(Tags.GraphQLSource, source);
+
+                // set analytics sample rate if enabled
+                var analyticsSampleRate = tracer.Settings.GetIntegrationAnalyticsSampleRate(IntegrationName, enabledWithGlobalSetting: false);
+                span.SetMetric(Tags.Analytics, analyticsSampleRate);
+            }
+            catch (Exception ex)
+            {
+                Log.ErrorException("Error creating or populating scope.", ex);
+            }
+
+            return scope;
+        }
+
+        private static Scope CreateScopeFromExecuteAsync(object executionContext)
+        {
+            if (!Tracer.Instance.Settings.IsIntegrationEnabled(IntegrationName))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            Tracer tracer = Tracer.Instance;
+            string source = executionContext.GetProperty("Document")
+                                            .GetProperty<string>("OriginalQuery")
+                                            .GetValueOrDefault();
+            string operationName = executionContext.GetProperty("Operation")
+                                                   .GetProperty<string>("Name")
+                                                   .GetValueOrDefault();
+            string operationType = executionContext.GetProperty("Operation")
+                                                   .GetProperty("OperationType")
+                                                   .ToString();
```

Comment:
You might want to use `as string` (and coalesce to `""` if needed) instead of `ToString()`. `MemberResult<string>.NotFound.ToString()` will probably return something like `MemberResult``1` unless we override `ToString()`.

Reply:
In 0a53d1489b804ba799cd881b948a6d3334254064 I changed the call to `executionContext.GetProperty("Operation").GetProperty<Enum>("OperationType").GetValueOrDefault().ToString();` . Does that work for you?

Reply:
Oops, I missed this. We can throw a NRE if we call `ToString()` on null.